variables.tf

```
@@ -37,6 +37,11 @@ variable "dependency_ids" {
   default = {}
 }
 
+variable"app_autosync" {
+  type = bool
+  default = true
+}
+
```

Wouldn't it be better if we pass a proper Terraform variable instead of using a boolean that activates a local variable?

```suggestion
variable "app_autosync" {
  type = map(object({
    allow_empty = bool
    prune = bool
    self_heal = bool
  }))
  default = {
    allow_empty = false
    prune = true
    self_heal = true
  }
}
```

The inconvenient with this approach is that in the event someone wants to deactivate the autosync is that they will be forced to declare the entire object in the module instantiation but at least it will be clearer. Otherwise as a bare minimum I would put a description in the variable saying what the variable activates in the locals. And I agree with @modridi, the variable should default as true, as I pointed out during the weekly (I do not know if you decided anything after I left).

---

After discussion, we'll leave this as-is for know, because the changes are urgently needed for Scaleway.